packages/go_router/lib/src/router.dart

```diff
@@ -509,6 +529,11 @@ class GoRouter implements RouterConfig<RouteMatchList> {
   String _effectiveInitialLocation(String? initialLocation) {
     final String platformDefault =
         WidgetsBinding.instance.platformDispatcher.defaultRouteName;
+    if (overridePlatformDefaultLocation) {
+      // can force null check as it's already verified by
+      // assert() while initialization
+      return initialLocation!;
+    }
```

Comment:
```suggestion
    if (overridePlatformDefaultLocation) {
      // can force null check as it's already verified by
      // assert() while initialization
      return initialLocation!;
    }
    final String platformDefault =
        WidgetsBinding.instance.platformDispatcher.defaultRouteName;
```

Reply:
The idea is to avoid declare the `platformDefault` if `overridePlatformDefaultLocation` is true.

Reply:
oh right, it'd definitely work! I'll commit, thanks.